ffmpeg: merge ffmpeg and ffmpeg-full

Atemu opened the pull request:
###### Description of changes

Fixes https://github.com/NixOS/nixpkgs/issues/15930

* [x] Merge Derivations
* [x] Refactor build flags
* [x] Reach build input conformance
* [x] Reach configureFlags conformance
* [x] Outputs conformance
* [x] Build/Install script conformance
* [x] ffmpeg_5 conformance
* [x] config.h conformance
* [x] Make headless variant work
* [x] Define "minimal", "headless" and "full"
  * [x] Should "full" have a lib output? It shouldn't be depended upon in Nixpkgs.
* [x] Remove asserts.
* [x] Ensure our we don't explicitly disable ffmpeg features that were implicitly enabled before (how?)
  * [x] Configure report
  * [x] Compare `-formats`, `pix_fmts`, etc.
* [x] Unify variants
  * [x] Multi-output
  * [x] Things not in ffmpeg-full
* [x] Untangle flags and dependencies
  * [x] Figure out to which flags the rest of the buildInputs belong
* [x] Cleanup
  * [x] Move code comments from dependencies to `with` flags
  * [x] Why `configurePlatforms = []`?
  * [x] Is the postFixup thingy still necessary?
* [x] Meta cleanup
  * [x] Make Licenses respond to flags
  * [x] Put maintainers back
* [ ] Compare closure sizes
* [x] Verify that cross-compilation still works
* [x] Do we want our rtmpdump or the one built in?

Wants:

* [ ] Put Darwin frameworks behind finer flags
* [ ] Restore commented flags
* [ ] Cleanup

Differences:

* headless variant's pname now reflects the variant
* Non-full variants have full's features explicitly disabled
* Avresample is disabled/removed
* nv-codec-headers are only included when necessary
* Implicitly enabled features are now explicitly enabled

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

r-ryantm committed:
stern: 1.21.0 -> 1.22.0

athre0z committed:
zydis: split `zycore` into separate derivation

This commit splits the Zycore dependency that was previously pulled
in via `fetchSubmodules` into a separate derivation. The package
previously didn't install the Zycore headers and CMake configs, making
it impossible to actually compile any code that uses Zydis because Zydis
includes Zycore headers.

athre0z committed:
zydis: add athre0z as a maintainer

Atemu committed:
ffmpeg: merge with ffmpeg-full

A ton of cleanup and refactoring went into this. Too much to list.

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>
Co-authored-by: Martin Weinelt <mweinelt@users.noreply.github.com>

Atemu commented:
ffmpeg-headless, ffmpeg and ffmpeg full build on my aarch64-linux systems now!

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,704 @@
+
+, ffmpegVariant ? "minimal" # Decides which dependencies are enabled by default
+
+  # Build with headless deps; excludes dependencies that are only necessary for
+  # GUI applications. To be used for purposes that don't generally need such
+  # components and i.e. only depend on libav
+, withHeadlessDeps ? ffmpegVariant == "headless" || withMinimalDeps
+
+  # Dependencies a user might customarily expect from a regular ffmpeg build.
+  # /All/ packages that depend on ffmpeg and some of its feaures should depend
+  # on the minimal variant. Minimal means the minimal set of features that
+  # satisfies all dependants in Nixpkgs
+, withMinimalDeps ? ffmpegVariant == "minimal" || withFullDeps
+
+  # Everything enabled; only guarded behind platform exclusivity or brokeness.
+  # If you need to depend on ffmpeg-full because ffmpeg is missing some feature
+  # your package needs, you should enable that feature in regular ffmpeg
+  # instead.
+, withFullDeps ? ffmpegVariant == "full"
```
Please let me know whether this is a sane policy.

SuperSandro2000 replied:
LGTM fine to me. If we notice it does not work out we can always change it later again.

One minor thing: we should assert that it is only one of the three strings we expect.

Artturin committed:
nixos/no-x-libs: add more replacements

all-packages: change -nox overrides to callPackage's so there is no
infinite recursion when replacing the package with a overlay

AndersonTorres commented:
> Define "minimal", "headless" and "full"

What about `small` instead of a polemical `minimal` attribute name?

> Should "full" have a lib output? 

`lib` should be an output by itself, mostly independent from `bin` files.

AndersonTorres commented on pkgs/top-level/all-packages.nix:
```diff
@@ -18975,30 +18975,27 @@ with pkgs;
-    inherit (darwin.apple_sdk.frameworks) Cocoa CoreMedia VideoToolbox;
-  };
+    # TODO still necessary?
+    # rtmpdump = null; # Prefer the built-in RTMP implementation
```
Following that discussion about splicing and avoiding `inherit ()` in all-packages, it would be more useful to control this with a `...Support` flag.

Atemu replied:
Yes absolutely, I'm just wondering whether we should still favour the built-in rtmp implementation.

Atemu replied:
I decided to just use ours.

YellowOnion commented:
Could we get libplacebo support as well?

Atemu commented:
Not in the scope of this PR.

It should certainly become easier to add support for more ffmpeg features with a cleaner derivation though.

YellowOnion commented:
Yeah I think I'll wait for this patch before attempting to add it, vulkan support on ffmpeg_4, ffmpeg_5 & ffmpeg_5Full seems to be broken, and needed for libplacebo.

r-ryantm committed:
igraph: 0.10.2 -> 0.10.3

Atemu committed:
jami: use regular ffmpeg_5

Our regular ffmpeg should have everything enabled Jami needs, if not that's a
bug we should fix in our ffmpeg rather than in jami

Atemu commented:
We're now at the point where the merge is mostly done with only some cleanup of the old ffmpeg-full derivation necessary but that could theoretically come later.

I'd love to have some more eyes on this.

Commits will probably be squashed in the end.

mweinelt commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,189 +1,673 @@
+ */
+, withGPL ? true # GPL: fdkaac,openssl,frei0r,cdio,samba,utvideo,vidstab,x265,x265,xavs,avid,zvbi,x11grab
+, withGPLv3 ? true # (L)GPL3: libvmaf,opencore-amrnb,opencore-amrwb,samba,vo-aacenc,vo-amrwbenc
+, withUnfree ? false # NONFREE: openssl,fdkaac,blackmagic-design-desktop-video
```
openssl 3.0 is asl20 licensed.

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -692,16 +692,10 @@ stdenv.mkDerivation rec {
+    license = with licenses; [ lgpl21Plus ]
+      ++ optional withGPL gpl2Plus
+      ++ optional withGPLv3 gpl3
+      ++ optional withUnfree unfreeRedistributable;
```
I don't see why. It's not like the lgpl code suddenly becomes GPL3 when you include GPL3 libraries.

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -334,8 +334,9 @@ stdenv.mkDerivation rec {
 
-  src = fetchurl {
-    url = "https://www.ffmpeg.org/releases/ffmpeg-${version}.tar.bz2";
+  src = fetchgit {
```
Yes, it is more desirable to build from the actual source not the source + whatever manual steps? + compressed into a tarball.

I've had to try a different rev at least once already.

Asked the other way around, why should we keep using the tarballs?

AndersonTorres replied:
> Asked the other way around, why should we keep using the tarballs?

First, I am not suggesting that tarballs should be mandatory. I am merely asking why the trending changed here specifically.

Second, usually the tarballs need less dependencies: a fetchurl/curl versus fetchgit/git to begin with.

Third, sometimes the tarballs are preprocessed in order to simplify the building. Indeed in Nixpkgs we have `autoreconfHook` because sources grabbed from Git regularly expose its building scripts.

And fourth, this is irrelevant, but since you ask, here is your answer.

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,486 @@
+ */
+, gplLicensing ? true # GPL: fdkaac,openssl,frei0r,cdio,samba,utvideo,vidstab,x265,x265,xavs,avid,zvbi,x11grab
+, version3Licensing ? true # (L)GPL3: libvmaf,opencore-amrnb,opencore-amrwb,samba,vo-aacenc,vo-amrwbenc
+, nonfreeLicensing ? false # NONFREE: openssl,fdkaac,blackmagic-design-desktop-video
```
No idea, that came from the original ffmpeg-full. I removed the comments already.

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,486 @@
+/*
+ *  External libraries options
+ */
+, alsa-lib ? null # Alsa in/output support
```
They're not which is why I already removed them.

For future reference, GH keeps your comments on old commits when you hold them for review.

AndersonTorres replied:
This is strange, because I was looking for the "changes from all commits". Maybe some heisenbug of Firefox...

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,486 @@
-, pulseaudioSupport ? stdenv.isLinux
-, libpulseaudio
-, ...
+, strippingDeveloper ? false
```
( ͡° ͜ʖ ͡°)

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,486 @@
+/*
+ *  External libraries
+ */
+#assert decklinkExtlib -> blackmagic-design-desktop-video != null
```
Yeah, original ffmpeg-full was a mess. They've all been removed since.

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,486 @@
+ *  Licensing dependencies
+ */
+assert version3Licensing -> gplLicensing;
+assert nonfreeLicensing -> gplLicensing && version3Licensing;
```
These were carried over from ffmpeg-full. I threw out a bunch of asserts already.

This one likely represents assertions inside the configure scripts. I'll take a look.

Atemu commented:
Also please note that the new ffmpeg derivation is based on the ffmpeg-full derivation: https://github.com/NixOS/nixpkgs/blob/356f88f80143ac3465b159a0db39ad38ea31abe7/pkgs/development/libraries/ffmpeg-full/default.nix

I've cleaned it up a lot but I don't plan on cleaning up everything till perfection; especially not in this PR.

mweinelt commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -212,9 +656,11 @@ stdenv.mkDerivation rec {
+      ++ optional withUnfree unfreeRedistributable;
     platforms = platforms.all;
-    inherit branch knownVulnerabilities;
+    maintainers = with maintainers; [ codyopel atemu ];
```
```suggestion
    maintainers = with maintainers; [ atemu ];
```

Last interaction with this package back in 2015/05. I think it's safe to say they moved on.

Atemu replied:
@codyopel

dotlambda committed:
python310Packages.igraph: 0.10.2 -> 0.10.3

https://github.com/igraph/python-igraph/releases/tag/0.10.3

FRidh commented on pkgs/development/libraries/dav1d/default.nix:
```diff
@@ -31,7 +31,7 @@ stdenv.mkDerivation rec {
   ];
 
-  doCheck = true;
+  doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
```
in `buildPythonPackage` we could also convert `true` to `stdenv.hostPlatform == stdenv.buildPlatform;`. I highly doubt we're going to see the use of an emulator in the `checkPhase` anytime soon.

Atemu replied:
If you don't know of many package whose checks can run on the buildPlatform, sure. Just revert my patch again in your PR then.

Atemu commented:
This is ready for a final round of reviews.

After that I'll squash and then it's ready to merge.

dotlambda committed:
python310Packages.pycaption: 1.0.1 -> 2.1.0

https://github.com/pbs/pycaption/blob/2.1.0/docs/changelog.rst

r-ryantm committed:
aoc-cli: 0.11.0 -> 0.12.0

r-ryantm committed:
hyfetch: 1.4.4 -> 1.4.6

rnhmjoj committed:
nixos-install: remove root requirement for bind mount

This moves the creation of the bind mount inside the `nixos-enter`
invocation. The command are executed in an unshared mount namespace, so
they can be run as an unprivileged user.

zowoq committed:
nixVersions.nix_2_13: init at 2.13.1

https://github.com/NixOS/nix/releases/tag/2.13.0
https://github.com/NixOS/nix/releases/tag/2.13.1

zowoq committed:
nixVersions.stable: 2.12 -> 2.13

zowoq committed:
nixos/nix-fallback-paths: 2.12.0 -> 2.13.1

zowoq committed:
nix-eval-jobs: pin to nix_2_12

Commit:
python3Packages.dm-tree: Fix build

r-ryantm committed:
katana: 0.0.2 -> 0.0.3

ProducerMatt committed:
rsbkb: init at 1.1

Plus Anderson's suggestions

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>

r-ryantm committed:
unpackerr: 0.10.1 -> 0.11.1

r-ryantm committed:
infracost: 0.10.15 -> 0.10.16

r-ryantm committed:
chamber: 2.11.0 -> 2.11.1

r-ryantm committed:
zinc: 0.3.5 -> 0.3.6

wegank committed:
bmon: unbreak on aarch64-darwin

bouk committed:
openmvs: init at 2.1.0

SuperSandro2000 commented on pkgs/tools/networking/unbound/default.nix:
```diff
@@ -110,7 +110,7 @@ stdenv.mkDerivation rec {
   checkInputs = [ bison ];
 
-  doCheck = true;
+  doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
```
```suggestion
  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
```

Atemu replied:
This doesn't actually work, oddly.

The manual on cross recommends what I did.

Changing this in future would be a simple search&replace if you have the motivation.

Artturin replied:
> The manual on cross recommends what I did.

fix in https://github.com/NixOS/nixpkgs/pull/211818

SuperSandro2000 commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,641 @@
+  buildInputs = optionals withFullDeps [
+    # TODO This was always in buildInputs before, why?
+    libdc1394
+  ] ++ optionals (withFullDeps && !stdenv.isDarwin) [
+    libraw1394
+  ]
```
```suggestion
  ] ++ optionals (withFullDeps && !stdenv.isDarwin) [ libraw1394 ]
```
nit

SuperSandro2000 commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,641 @@
-  postPatch = "patchShebangs .";
-  inherit patches;
+  postPatch = ''
+    patchShebangs .
```
Could we narrow this down to maybe one or two directories?

Atemu replied:
It takes like a second; I don't think it's worth figuring out which dirs need to be patched.

SuperSandro2000 commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -1,205 +1,641 @@
+/*
+ *  Licensing dependencies
+ */
+assert withGPLv3 -> withGPL;
+assert withUnfree -> withGPL && withGPLv3;
+/*
+ *  Build dependencies
+ */
+assert withPixelutils -> buildAvutil;
+/*
+ *  Program dependencies
+ */
+assert buildFfmpeg -> buildAvcodec
+                     && buildAvfilter
+                     && buildAvformat
+                     && buildSwresample;
+assert buildFfplay -> buildAvcodec
+                     && buildAvformat
+                     && buildSwscale
+                     && buildSwresample;
+assert buildFfprobe -> buildAvcodec && buildAvformat;
+/*
+ *  Library dependencies
+ */
+assert buildAvcodec -> buildAvutil; # configure flag since 0.6
+assert buildAvdevice -> buildAvformat
+                       && buildAvcodec
+                       && buildAvutil; # configure flag since 0.6
+assert buildAvformat -> buildAvcodec && buildAvutil; # configure flag since 0.6
+assert buildPostproc -> buildAvutil;
+assert buildSwscale -> buildAvutil;
```
👍🏼

Atemu replied:
Note: Not my work. These are ported from ffmpeg-full. I've trimmed them quite a bit.

SuperSandro2000 reviewed (commented):
Especially the conditions look really solid. Thanks for cleaning this big mess up! I appreciate it!

alyssais committed:
valgrind-light: fix build for FreeBSD

This is very similar to what we do for Darwin directly below, but
we look up the FreeBSD version at eval time instead.

The .0 is prepended because our current FreeBSD system doubles only
include the major version.  Valgrind ignores everything after the
dot anyway (but it wants the dot to be there).

alyssais committed:
lib.platforms.armv7: init

alyssais committed:
lib.platforms.power: init

alyssais committed:
lib.platforms.s390x: init

alyssais committed:
valgrind: make meta.platforms more accurate

Valgrind has to do lots of architecture- and OS-specific stuff, so it
only works on a defined set of platforms.  So rather than marking
platforms as bad one by one, let's set meta.platforms to the actual
set of platforms supported by Valgrind, as listed in its configure
script.

r-ryantm committed:
carapace: 0.19.1 -> 0.20.2

r-ryantm committed:
checkSSLCert: 2.57.0 -> 2.58.0

mweinelt committed:
Merge pull request #211210 from NixOS/home-assistant

home-assistant: 2023.1.4 -> 2023.1.5

rnhmjoj committed:
nixos/systemd-boot: improve initrd-secrets error messages

This is the equivalent of eccc1e5b but for systemd-boot

techknowlogick committed:
kubernetes-helm: enable fish completion

prusnak committed:
torq: 0.16.9 -> 0.16.15

dotlambda committed:
python310Packages.ytmusicapi: 0.24.1 -> 0.25.0

https://github.com/sigma67/ytmusicapi/releases/tag/0.25.0

raphaelr committed:
steam-run: add libxcrypt to FHS environment

Required by some games.

Kiskae committed:
linuxPackages.nvidia_x11_production: 525.78.01 -> 525.85.05

figsoda committed:
onefetch: 2.14.2 -> 2.15.1

Diff: https://github.com/o2sh/onefetch/compare/2.14.2...2.15.1

Changelog: https://github.com/o2sh/onefetch/blob/v2.15.1/CHANGELOG.md

r-ryantm committed:
erlangR25: 25.2 -> 25.2.1

rnhmjoj committed:
Revert "Revert "nixos/tests/installer: test relative paths in initrd secrets""

This reverts commit 8d45d82c71b91872e853f0bce3ed69993508ec5e.

The tests shold pass now that systemd-boot handles failures in initrd
for older boot entries.

dotlambda committed:
python310Packages.timezonefinder: update dependencies

dotlambda committed:
python310Packages.timezonefinder: add meta.changelog

Ericson2314 committed:
Merge pull request #211622 from alyssais/valgrind-platforms

valgrind: make meta.platforms more accurate

AndersonTorres committed:
Merge pull request #211487 from ProducerMatt/submit/rsbkb

rsbkb: init at 1.1

prusnak committed:
Merge pull request #211632 from prusnak/torq

torq: 0.16.9 -> 0.16.15

figsoda committed:
felix-fm: 2.2.2 -> 2.2.3

Diff: https://github.com/kyoheiu/felix/compare/v2.2.2...v2.2.3

Changelog: https://github.com/kyoheiu/felix/blob/v2.2.3/CHANGELOG.md

ElnuDev committed:
pureref: init at 1.11.1

r-ryantm committed:
fvwm3: 1.0.5 -> 1.0.6a

dotlambda committed:
python310Packages.johnnycanencrypt: 0.11.0 -> 0.12.0

https://github.com/kushaldas/johnnycanencrypt/releases/tag/v0.12.0

aanderse committed:
Merge pull request #211611 from newAM/github-runner

github-runner: 2.300.2 -> 2.301.1

aanderse committed:
Merge pull request #201836 from aanderse/kodi.packages

kodi: addon updates

nat-418 committed:
vimPlugins: update

alyssais committed:
bintools: add isGNU and isLLVM attributes

alyssais committed:
gcc: use as(1) from binutils with LLVM bintools

LLVM does not provide a drop-in replacement for as(1).

This makes it possible to build a GNU Fortran cross compiler from GNU
to LLVM — e.g. buildPackages.gfortran for
{ system = "aarch64-linux"; useLLVM = true; }

nat-418 committed:
vimPlugins.oil-nvim: init at 2023-01-19

r-ryantm committed:
klaus: 2.0.1 -> 2.0.2

bcardiff committed:
souffle: add darwin support (#208373)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

happysalada committed:
lighthouse: fix darwin

happysalada committed:
vector: 0.26.0 -> 0.27.0

AndersonTorres committed:
Merge pull request #211508 from ElnuDev/pureref

pureref: init at 1.11.1

r-ryantm committed:
git-trim: 0.4.2 -> 0.4.4

figsoda committed:
Merge pull request #211475 from figsoda/texture

texture-synthesis: fix build on aarch64-darwin

figsoda committed:
Merge pull request #211668 from nat-418/oil.nvim

vimPlugins: add oil.nvim

r-ryantm committed:
coder: 0.13.6 -> 0.14.3

0x4A6F committed:
Merge pull request #211620 from fgaz/lgogdownloader/gui

lgogdownloader: enable GUI

figsoda committed:
Merge pull request #211481 from figsoda/pw-viz

pw-viz: fix build

figsoda committed:
Merge pull request #211484 from figsoda/cargo-release

cargo-release: fix build, add figsoda as a maintainer

figsoda committed:
cargo-lock: 8.0.2 -> 8.0.3

Changelog: https://github.com/rustsec/rustsec/blob/cargo-lock/v8.0.3/cargo-lock/CHANGELOG.md

figsoda committed:
Merge pull request #211486 from figsoda/macchina

macchina: fix darwin build, add figsoda as a maintainer

figsoda committed:
Merge pull request #211489 from figsoda/ion

ion: unstable-2021-05-10 -> unstable-2022-11-27

NickCao committed:
Merge pull request #211546 from r-ryantm/auto-update/frr

frr: 8.4.1 -> 8.4.2

NickCao committed:
Merge pull request #211431 from chuangzhu/tangram

tangram: 1.3.2 -> 2.0

NickCao committed:
Merge pull request #211370 from r-ryantm/auto-update/tz

tz: 0.6.2 -> 0.6.3

NickCao committed:
Merge pull request #211365 from chuangzhu/blackbox

blackbox: 0.12.2 -> 0.13.1

NickCao committed:
Merge pull request #211363 from r-ryantm/auto-update/trilium-desktop

trilium-desktop: 0.58.5 -> 0.58.7

Artturin committed:
Merge pull request #211574 from Artturin/remacreintol

treewide: remove accidentally introduced -l

NickCao committed:
Merge pull request #211322 from r-ryantm/auto-update/okteto

okteto: 2.11.0 -> 2.11.1

r-ryantm committed:
sftpgo: 2.4.2 -> 2.4.3

0x4A6F committed:
Merge pull request #211660 from dotlambda/johnnycanencrypt-0.12.0

python310Packages.johnnycanencrypt: 0.11.0 -> 0.12.0

mweinelt committed:
python3Packages.dm-env: Fix tests

roberth committed:
Merge pull request #210976 from clhodapp/fix/extensible-sets

lib: make extender available on self-references

trofi committed:
Merge pull request #211136 from r-ryantm/auto-update/socat

socat: 1.7.4.3 -> 1.7.4.4

mweinelt committed:
Merge pull request #211301 from ziguana/master

python3Packages.dm-tree: Fix build

r-ryantm committed:
trufflehog: 3.24.0 -> 3.25.2

nat-418 committed:
vimPlugins: update

nat-418 committed:
vimPlugins.dirbuf-nvim: init at 2022-08-28

NickCao committed:
Merge pull request #208798 from squalus/standardnotes

standardnotes: 3.23.69 -> 3.129.0

NickCao committed:
Merge pull request #200943 from r-ryantm/auto-update/python310Packages.dropbox

python310Packages.dropbox: 11.35.0 -> 11.36.0

NickCao committed:
Merge pull request #188795 from Sohalt/oci-containers-wait-for-network

nixos/oci-containers: wait for network before starting container

rodrgz committed:
bitwarden-menu: init at 0.4.1

NickCao committed:
Merge pull request #211541 from r-ryantm/auto-update/python310Packages.azure-mgmt-compute

python310Packages.azure-mgmt-compute: 29.0.0 -> 29.1.0

NickCao committed:
Merge pull request #209593 from JamieMagee/azure-functions-core-tools-4.0.4915

azure-functions-core-tools: 4.0.4785 -> 4.0.4915

techknowlogick committed:
gitea: 1.18.1 -> 1.18.2

r-ryantm committed:
strawberry: 1.0.13 -> 1.0.14

happysalada committed:
erigon: 2.34.0 -> 2.35.2

starcraft66 committed:
hubble: mark not broken on darwin

figsoda committed:
Merge pull request #211688 from nat-418/dirbuf.nvim

vimPlugins: add dirbuf.nvim

r-ryantm committed:
python310Packages.flake8-bugbear: 22.12.6 -> 23.1.17

r-ryantm committed:
python310Packages.google-cloud-org-policy: 1.5.0 -> 1.7.0

github-actions[bot] committed:
terraform-providers.alicloud: 1.196.0 → 1.197.0

MostAwesomeDude committed:
Explain why unit tests are disabled.

This was requested during code review.

github-actions[bot] committed:
terraform-providers.azuread: 2.32.0 → 2.33.0

github-actions[bot] committed:
terraform-providers.newrelic: 3.12.0 → 3.13.0

github-actions[bot] committed:
terraform-providers.sentry: 0.11.1 → 0.11.2

github-actions[bot] committed:
terraform-providers.aws: 4.50.0 → 4.51.0

github-actions[bot] committed:
terraform-providers.spotinst: 1.94.0 → 1.95.1

NickCao committed:
Merge pull request #211537 from NickCao/sure

python3Packages.sure: disable test on python 3.11 or later

NickCao committed:
Merge pull request #211269 from MostAwesomeDude/yark-init

python3Packages.yark: init at 1.2.3

bobby285271 committed:
Merge pull request #211086 from bobby285271/upd/xneur-enchant2

xneur: build with enchant2, enchant1: remove

bcdarwin committed:
python310Packages.dcmstack: 0.8 -> 0.9; unbreak

r-ryantm committed:
python310Packages.flask-restx: 1.0.3 -> 1.0.5

figsoda committed:
ruff: 0.0.226 -> 0.0.227

Diff: https://github.com/charliermarsh/ruff/compare/v0.0.226...v0.0.227

Changelog: https://github.com/charliermarsh/ruff/releases/tag/v0.0.227

bobby285271 committed:
Merge pull request #211346 from bobby285271/upd/evolution

evolution: 3.46.2 → 3.46.3

zowoq committed:
nix-direnv: 2.2.0 -> 2.2.1

Diff: https://github.com/nix-community/nix-direnv/compare/2.2.0...2.2.1

zowoq committed:
kubernetes: 1.26.0 -> 1.26.1

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md#changelog-since-v1260

bcdarwin committed:
python310Packages.pylibjpeg-libjpeg: init at 1.3.3

bobby285271 committed:
Merge pull request #211495 from bobby285271/upd/cinnamon

cinnamon.nemo: 5.6.2 -> 5.6.3

NickCao committed:
Merge pull request #211635 from Kiskae/patch-4

linuxPackages.nvidia_x11_production: 525.78.01 -> 525.85.05

Mic92 committed:
Merge pull request #211711 from qowoz/nix-direnv

nix-direnv: 2.2.0 -> 2.2.1

NickCao committed:
linuxPackages.cryptodev: cleanup and fix cross compilation

r-ryantm committed:
ddosify: 0.11.0 -> 0.12.0

r-ryantm committed:
mdbook-mermaid: 0.12.1 -> 0.12.6

collares committed:
sage: backport intermittent test fix

NickCao committed:
vsmtp: fix build

bobby285271 committed:
Merge pull request #211564 from datafoo/vscode-extensions.elixir-lsp.vscode-elixir-ls

vscode-extensions.elixir-lsp.vscode-elixir-ls: 0.12.0 -> 0.13.0

NickCao committed:
python3Packages.cryptg: fix build on darwin

NickCao committed:
Merge pull request #211702 from r-ryantm/auto-update/python310Packages.flake8-bugbear

python310Packages.flake8-bugbear: 22.12.6 -> 23.1.17

NickCao committed:
Merge pull request #211629 from techknowlogick/helm-fish

kubernetes-helm: enable fish completion

NickCao committed:
Merge pull request #211588 from ilya-fedin/kotato-fix-darwin

kotatogram-desktop: fix build on Darwin after #210062

NickCao committed:
Merge pull request #211579 from fabaff/pontos-bump

python310Packages.pontos: 23.1.1 -> 23.1.3

NickCao committed:
Merge pull request #211570 from r-ryantm/auto-update/cinny-desktop

cinny-desktop: 2.2.2 -> 2.2.3

NickCao committed:
Merge pull request #211554 from r-ryantm/auto-update/python310Packages.thespian

python310Packages.thespian: 3.10.6 -> 3.10.7

NickCao committed:
Merge pull request #211493 from r-ryantm/auto-update/eksctl

eksctl: 0.124.0 -> 0.125.0

NickCao committed:
Merge pull request #211453 from fabaff/angr-bump

python310Packages.angr: 9.2.33 -> 9.2.34

NickCao committed:
Merge pull request #211405 from devholic/upbound-0.15.0

upbound: 0.12.2 -> 0.15.0

NickCao committed:
Merge pull request #211349 from mislavzanic/init/terminal-stocks

terminal-stocks: init at 1.0.14

NickCao committed:
Merge pull request #211088 from r-ryantm/auto-update/python310Packages.rq

python310Packages.rq: 1.11.1 -> 1.12.0

NickCao committed:
Merge pull request #211350 from NixOS/revert-211291-revert-211204-gajim-1.6.1

Revert "Revert "pythonPackages.nbxmpp: 4.0.0 → 4.0.1; gajim: 1.6.0 → 1.6.1""

r-ryantm committed:
python310Packages.pyoctoprintapi: 0.1.10 -> 0.1.11

NickCao committed:
Merge pull request #211729 from NickCao/cryptg

python3Packages.cryptg: fix build on darwin

NickCao committed:
Merge pull request #211726 from NickCao/vsmtp

vsmtp: fix build

NickCao committed:
Merge pull request #211479 from fabaff/pubnub-bump

python310Packages.pubnub: 7.0.2 -> 7.1.0

NickCao committed:
Merge pull request #211465 from fabaff/total-connect-client-bump

python310Packages.total-connect-client: 2022.10 -> 2023.1

NickCao committed:
Merge pull request #210631 from Majiir/fix-octoprint-overlays

octoprint: fix overlay composition

NickCao committed:
Merge pull request #210215 from kilianar/xdot-gtk

xdot: fix Gtk error

collares committed:
Merge pull request #211725 from collares/sage-expect

sage: backport intermittent test fix

fabaff committed:
trufflehog: specify license

r-ryantm committed:
adguardhome: 0.107.21 -> 0.107.22

fabaff committed:
Merge pull request #211643 from dotlambda/timezonefinder-fix

python310Packages.timezonefinder: update dependencies

fabaff committed:
Merge pull request #211722 from r-ryantm/auto-update/ddosify

ddosify: 0.11.0 -> 0.12.0

r-ryantm committed:
python310Packages.python-snap7: 1.2 -> 1.3

fabaff committed:
Merge pull request #211731 from r-ryantm/auto-update/python310Packages.pyoctoprintapi

python310Packages.pyoctoprintapi: 0.1.10 -> 0.1.11

fabaff committed:
Merge pull request #211578 from fabaff/qovery-cli-bump

qovery-cli: 0.48.3 -> 0.48.4

trofi committed:
Merge pull request #209268 from toastal/lightningcss-1.18.0

lightningcss: 0.17.1 → 0.18.0

fabaff committed:
python310Packages.pysoma: 0.0.12 -> 0.0.13

fabaff committed:
python310Packages.atenpdu: 0.4.0 -> 0.5.0

fabaff committed:
python310Packages.asysocks: 0.2.3 -> 0.2.5

Changelog: https://github.com/skelsec/asysocks/releases/tag/0.2.5

fabaff committed:
python310Packages.asyauth: 0.0.10 -> 0.0.11

Changelog: https://github.com/skelsec/asyauth/releases/tag/0.0.11

fabaff committed:
memray: 1.5.0 -> 1.6.0

Diff: https://github.com/bloomberg/memray/compare/refs/tags/v1.5.0...v1.6.0

Changelog: https://github.com/bloomberg/memray/releases/tag/v1.6.0

bryanasdev000 committed:
jrnl: 3.0 -> 3.3

- fix build

- drop tzlocal and add rich replaces

- add myself as maintainer

pSub committed:
Merge pull request #211659 from r-ryantm/auto-update/klaus

klaus: 2.0.1 -> 2.0.2

r-ryantm committed:
python310Packages.zope_filerepresentation: 5.0.0 -> 6.0

bobby285271 committed:
Merge pull request #211563 from datafoo/vscode-extensions.davidanson.vscode-markdownlint

vscode-extensions.davidanson.vscode-markdownlint: 0.48.1 -> 0.49.0

bobby285271 committed:
Merge pull request #211562 from datafoo/vscode-extensions.esbenp.prettier-vscode

vscode-extensions.esbenp.prettier-vscode: 9.10.3 -> 9.10.4

marsam committed:
Merge pull request #211669 from r-ryantm/auto-update/git-trim

git-trim: 0.4.2 -> 0.4.4

fabaff committed:
Merge pull request #211739 from fabaff/memray-bump

memray: 1.5.0 -> 1.6.0

fabaff committed:
Merge pull request #211738 from fabaff/asyauth-bump

python310Packages.asysocks: 0.2.3 -> 0.2.5, python310Packages.asyauth: 0.0.10 -> 0.0.11

fabaff committed:
Merge pull request #211737 from fabaff/atenpdu-bump

python310Packages.atenpdu: 0.4.0 -> 0.5.0

fabaff committed:
Merge pull request #211736 from fabaff/pysoma-bump

python310Packages.pysoma: 0.0.12 -> 0.0.13

r-ryantm committed:
python310Packages.bottleneck: 1.3.5 -> 1.3.6

flokli committed:
Merge pull request #211576 from r-ryantm/auto-update/python310Packages.bpython

python310Packages.bpython: 0.23 -> 0.24

fabaff committed:
python310Packages.odp-amsterdam: 5.0.0 -> 5.0.1

Diff: https://github.com/klaasnicolaas/python-odp-amsterdam/compare/refs/tags/v5.0.0...v5.0.1

fabaff committed:
python310Packages.odp-amsterdam: update meta

aaronjheng committed:
d2: 0.1.5 -> 0.1.6

afh committed:
got: 0.79 -> 0.81

domenkozar committed:
Merge pull request #195606 from geopjr-forks/update/crystal

crystal: 1.2 -> 1.7

NickCao committed:
Merge pull request #211719 from NickCao/cryptodev

linuxPackages.cryptodev: cleanup and fix cross compilation

SuperSandro2000 committed:
Merge pull request #211760 from afh/afh-got-update

SuperSandro2000 committed:
Merge pull request #204572 from r-ryantm/auto-update/seer

SuperSandro2000 committed:
Merge pull request #211757 from aaronjheng/d2

SuperSandro2000 committed:
Merge pull request #211752 from r-ryantm/auto-update/python310Packages.bottleneck

SuperSandro2000 committed:
Merge pull request #211616 from figsoda/cargo-generate

bouk committed:
openmvg: 2.0 -> unstable-2022-12-30 (#211560)

GenericNerdyUsername committed:
python310Packages.edalize: fix build (#211601)

thiagokokada committed:
Merge pull request #209060 from Sohalt/graal-vm-js

pkgs/graalvm: add graaljs installable

SuperSandro2000 committed:
Merge pull request #211735 from r-ryantm/auto-update/python310Packages.python-snap7

SuperSandro2000 committed:
Merge pull request #211679 from r-ryantm/auto-update/sftpgo

SuperSandro2000 committed:
Merge pull request #211740 from bryanasdev000/jrnl33

SuperSandro2000 committed:
Merge pull request #211686 from r-ryantm/auto-update/trufflehog

SuperSandro2000 committed:
Merge pull request #211744 from r-ryantm/auto-update/python310Packages.zope_filerepresentation

dantefromhell committed:
factorio-experimental: 1.1.74 -> 1.1.76 (#211452)

Co-authored-by: dantefromhell <dantefromhell@users.noreply.github.com>

SuperSandro2000 committed:
Merge pull request #211708 from r-ryantm/auto-update/python310Packages.flask-restx

SuperSandro2000 committed:
Merge pull request #211709 from figsoda/ruff

SuperSandro2000 committed:
Merge pull request #206502 from jchw-forks/update/ruffle

SuperSandro2000 committed:
Merge pull request #211714 from bcdarwin/update-python3-dcmstack

7c6f434c committed:
Merge pull request #210443 from r-ryantm/auto-update/apache-jena

apache-jena: 4.6.1 -> 4.7.0

SuperSandro2000 committed:
Merge pull request #211699 from starcraft66/patch-2

SuperSandro2000 committed:
Merge pull request #211704 from r-ryantm/auto-update/python310Packages.google-cloud-org-policy

7c6f434c committed:
Merge pull request #210754 from r-ryantm/auto-update/fdm

fdm: 2.1 -> 2.2

SuperSandro2000 committed:
Merge pull request #211692 from rodrgz/bwm

SuperSandro2000 committed:
Merge pull request #211678 from figsoda/cargo-lock

SuperSandro2000 committed:
Merge pull request #208739 from gbtb/gsmartcontrol_114

SuperSandro2000 committed:
Merge pull request #204287 from Artturin/addmorenox

SuperSandro2000 committed:
Merge pull request #211658 from raphaelr/steam-libxcrypt

7c6f434c committed:
Merge pull request #211245 from onny/firejail-update

firejail: 0.9.70 -> 0.9.72

7c6f434c committed:
Merge pull request #211041 from r-ryantm/auto-update/ogre

ogre: 13.5.3 -> 13.6.0

SuperSandro2000 committed:
Merge pull request #211653 from figsoda/felix

SuperSandro2000 committed:
Merge pull request #211636 from figsoda/onefetch

SuperSandro2000 committed:
Merge pull request #211633 from dotlambda/ytmusicapi-0.25.0

SuperSandro2000 committed:
Merge pull request #211628 from Trundle/pulumi-azure-native-1.92.0

SuperSandro2000 committed:
Merge pull request #208804 from justinas/teleport-11

SuperSandro2000 committed:
Merge pull request #203102 from r-ryantm/auto-update/python310Packages.docplex

SuperSandro2000 committed:
Merge pull request #210544 from wegank/poedit-bump

SuperSandro2000 committed:
Merge pull request #211605 from SFrijters/xmldiff-2.5

SuperSandro2000 committed:
Merge pull request #198594 from r-ryantm/auto-update/python310Packages.oci

SuperSandro2000 committed:
Merge pull request #211595 from r-ryantm/auto-update/python310Packages.textual

SuperSandro2000 committed:
Merge pull request #211582 from r-ryantm/auto-update/python310Packages.pulumi-aws

SuperSandro2000 committed:
Merge pull request #211590 from firefly-cpp/tcxreader-044

SuperSandro2000 committed:
Merge pull request #211568 from Luflosi/update/raspberrypi-eeprom

SuperSandro2000 committed:
Merge pull request #210506 from uninsane/pr/whalebird-4.7.4

fpletz committed:
Merge pull request #211194 from fpletz/pkgs/libvirt-9.0.0

SuperSandro2000 committed:
Merge pull request #211572 from bouk/bouk/update-vcg

SuperSandro2000 committed:
Merge pull request #211566 from r-ryantm/auto-update/python310Packages.moderngl-window

SuperSandro2000 committed:
Merge pull request #211555 from fgaz/pikchr/2022-12-07

SuperSandro2000 committed:
Merge pull request #211539 from r-ryantm/auto-update/python310Packages.libcloud

yuyuyureka committed:
gccgo: wrap go binary (#207670)

yayayayaka committed:
gitlab-runner: 15.7.2 -> 15.8.0 (#211591)

https://gitlab.com/gitlab-org/gitlab-runner/blob/v15.8.0/CHANGELOG.md

ajs124 committed:
Merge pull request #198166 from helsinki-systems/upd/open-iscsi

openiscsi: 2.1.7 -> 2.1.8

ajs124 committed:
Merge pull request #211031 from r-ryantm/auto-update/unpoller

unpoller: 2.4.1 -> 2.7.10

SuperSandro2000 committed:
Merge pull request #211734 from r-ryantm/auto-update/adguardhome

ajs124 committed:
Merge pull request #205481 from SFrijters/authy-2.2.2

authy: 2.2.1 -> 2.2.2

marsam committed:
Merge pull request #208410 from r-ryantm/auto-update/cargo-watch

cargo-watch: 8.1.2 -> 8.3.0

rnhmjoj committed:
pdns-recursor: 4.8.0 -> 4.8.1

rnhmjoj committed:
Merge pull request #211779 from rnhmjoj/pr-pdns

pdns-recursor: 4.8.0 -> 4.8.1

newAM committed:
cargo-spellcheck: 0.12.2 -> 0.12.3

fabaff committed:
Merge pull request #211756 from fabaff/odp-amsterdam-bump

python310Packages.odp-amsterdam: 5.0.0 -> 5.0.1

Ma27 committed:
Merge pull request #211694 from techknowlogick/gitea-1182

gitea: 1.18.1 -> 1.18.2

yuyuyureka committed:
pkgsMusl.go: bootstrap with gccgo (#207681)

7c6f434c committed:
Merge pull request #211780 from bouk/bouk/openmvs

openmvs: init at 2.1.0

rnhmjoj committed:
Merge pull request #211295 from rnhmjoj/pr-nixos-install

nixos-install: remove root requirement for bind mount

fabaff committed:
Merge pull request #211621 from r-ryantm/auto-update/checkSSLCert

checkSSLCert: 2.57.0 -> 2.58.0

r-ryantm committed:
erlang: 24.3.4.7 -> 24.3.4.8

SuperSandro2000 committed:
Merge pull request #210474 from linsui/i2p

i2p: 1.9.0 -> 2.1.0

Atemu committed:
handbrake: disable sdl2 in custom ffmpeg

Fixes https://github.com/NixOS/nixpkgs/issues/211297

Artturin commented on pkgs/tools/networking/unbound/default.nix:
```diff
@@ -110,7 +110,7 @@ stdenv.mkDerivation rec {
   checkInputs = [ bison ];
 
-  doCheck = true;
+  doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
```
why is this cross commit in this PR?

these changes also aren't necessary https://github.com/NixOS/nixpkgs/blob/c4961615c7b65e4186076ae3bf521eb40ad48f91/pkgs/stdenv/generic/make-derivation.nix#L168

Atemu replied:
The snippet above is broken. I needed to add this because unbound wouldn't build.

Atemu replied:
Try `with pkgsCross.armv7l-hf-multiplatform.stdenv; buildPlatform.canExecute hostPlatform` in a repl.

Atemu replied:
(The reason it's here is because I wanted to validate cross still works.)

Artturin replied:
`pkgsCross.aarch64-multiplatform.unbound` and so does `pkgsCross.armv7l-hf-multiplatform.unbound` from x86_64-linux

Atemu replied:
`nix-build -A pkgsCross.armv7l-hf-multiplatform.unbound` after disabling doCheck on nettle since that's now broken too:

    error: builder for '/nix/store/fvxq7hkrwj05k0dg7jl5n24fqjm9p7mq-unbound-armv7l-unknown-linux-gnueabihf-1.17.0.drv' failed with exit code 2;
           last 10 log lines:
           > libtool: compile:  armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -c testcode/testbound.c  -fPIC -DPIC -o .libs/testbound.o
           > ./libtool --tag=CC --mode=compile armv7l-unknown-linux-gnueabihf-gcc -I.  -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE  -o replay.lo -c testcode/replay.c
           > libtool: compile:  armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -c testcode/replay.c  -fPIC -DPIC -o .libs/replay.o
           > ./libtool --tag=CC --mode=compile armv7l-unknown-linux-gnueabihf-gcc -I.  -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE  -o fake_event.lo -c testcode/fake_event.c
           > libtool: compile:  armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -c testcode/fake_event.c  -fPIC -DPIC -o .libs/fake_event.o
           > ./libtool --tag=CC --mode=link armv7l-unknown-linux-gnueabihf-gcc  -R/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -R/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib -I.  -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE -pie -Wl,-z,relro,-z,now -L/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -L/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib -L/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/lib -o testbound testbound.lo replay.lo fake_event.lo testpkts.lo worker.lo acl_list.lo daemon.lo stats.lo shm_main.lo dns.lo infra.lo rrset.lo dname.lo msgencode.lo as112.lo msgparse.lo msgreply.lo packed_rrset.lo iterator.lo iter_delegpt.lo iter_donotq.lo iter_fwd.lo iter_hints.lo iter_priv.lo iter_resptype.lo iter_scrub.lo iter_utils.lo localzone.lo mesh.lo modstack.lo view.lo outbound_list.lo alloc.lo config_file.lo configlexer.lo configparser.lo fptr_wlist.lo edns.lo locks.lo log.lo mini_event.lo module.lo net_help.lo random.lo rbtree.lo regional.lo rtt.lo dnstree.lo lookup3.lo lruhash.lo slabhash.lo tcp_conn_limit.lo timehist.lo tube.lo winsock_event.lo autotrust.lo val_anchor.lo rpz.lo proxy_protocol.lo validator.lo val_kcache.lo val_kentry.lo val_neg.lo val_nsec3.lo val_nsec.lo val_secalgo.lo val_sigcrypt.lo val_utils.lo dns64.lo  authzone.lo         respip.lo ub_event.lo keyraw.lo sbuffer.lo wire2str.lo parse.lo parseutil.lo rrdef.lo str2wire.lo memcmp.lo malloc.lo strlcat.lo strlcpy.lo arc4random.lo arc4random_uniform.lo arc4_lock.lo -lssl -levent  -lcrypto
           > libtool: link: armv7l-unknown-linux-gnueabihf-gcc -I. -I/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/include -I/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/include -I/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/include -DSRCDIR=. -g -O2 -flto -fPIE -pie -Wl,-z -Wl,relro -Wl,-z -Wl,now -o testbound .libs/testbound.o .libs/replay.o .libs/fake_event.o .libs/testpkts.o .libs/worker.o .libs/acl_list.o .libs/daemon.o .libs/stats.o .libs/shm_main.o .libs/dns.o .libs/infra.o .libs/rrset.o .libs/dname.o .libs/msgencode.o .libs/as112.o .libs/msgparse.o .libs/msgreply.o .libs/packed_rrset.o .libs/iterator.o .libs/iter_delegpt.o .libs/iter_donotq.o .libs/iter_fwd.o .libs/iter_hints.o .libs/iter_priv.o .libs/iter_resptype.o .libs/iter_scrub.o .libs/iter_utils.o .libs/localzone.o .libs/mesh.o .libs/modstack.o .libs/view.o .libs/outbound_list.o .libs/alloc.o .libs/config_file.o .libs/configlexer.o .libs/configparser.o .libs/fptr_wlist.o .libs/edns.o .libs/locks.o .libs/log.o .libs/mini_event.o .libs/module.o .libs/net_help.o .libs/random.o .libs/rbtree.o .libs/regional.o .libs/rtt.o .libs/dnstree.o .libs/lookup3.o .libs/lruhash.o .libs/slabhash.o .libs/tcp_conn_limit.o .libs/timehist.o .libs/tube.o .libs/winsock_event.o .libs/autotrust.o .libs/val_anchor.o .libs/rpz.o .libs/proxy_protocol.o .libs/validator.o .libs/val_kcache.o .libs/val_kentry.o .libs/val_neg.o .libs/val_nsec3.o .libs/val_nsec.o .libs/val_secalgo.o .libs/val_sigcrypt.o .libs/val_utils.o .libs/dns64.o .libs/authzone.o .libs/respip.o .libs/ub_event.o .libs/keyraw.o .libs/sbuffer.o .libs/wire2str.o .libs/parse.o .libs/parseutil.o .libs/rrdef.o .libs/str2wire.o .libs/memcmp.o .libs/malloc.o .libs/strlcat.o .libs/strlcpy.o .libs/arc4random.o .libs/arc4random_uniform.o .libs/arc4_lock.o  -L/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -L/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib -L/nix/store/83463yq115sycy9vcq5q32abbih91gc0-expat-armv7l-unknown-linux-gnueabihf-2.5.0-dev/lib -lssl /nix/store/w771s40v542jmi4hyq2l3v0y96xxxia7-libevent-armv7l-unknown-linux-gnueabihf-2.1.12/lib/libevent.so -lcrypto -Wl,-rpath -Wl,/nix/store/w771s40v542jmi4hyq2l3v0y96xxxia7-libevent-armv7l-unknown-linux-gnueabihf-2.1.12/lib -Wl,-rpath -Wl,/nix/store/w771s40v542jmi4hyq2l3v0y96xxxia7-libevent-armv7l-unknown-linux-gnueabihf-2.1.12/lib -Wl,-rpath -Wl,/nix/store/fgwn7r8ihkmlhcxsph9dwf2p71wkbv19-openssl-armv7l-unknown-linux-gnueabihf-3.0.7-dev/lib -Wl,-rpath -Wl,/nix/store/0l9hp4hn50h3sfjn91rxynsmfb120zw0-libevent-armv7l-unknown-linux-gnueabihf-2.1.12-dev/lib
           > ./unittest
           > /nix/store/sc8jyzxrzkw4sq01fa9zbgj530prz2sy-bash-5.2-p15/bin/bash: line 1: ./unittest: cannot execute binary file: Exec format error
           > make: *** [Makefile:345: test] Error 126
           For full logs, run 'nix log /nix/store/fvxq7hkrwj05k0dg7jl5n24fqjm9p7mq-unbound-armv7l-unknown-linux-gnueabihf-1.17.0.drv'.

Atemu replied:
Took the cross fixes out; that can come at a later day.

figsoda committed:
Merge pull request #211790 from newAM/cargo-spellcheck-0.12.3

cargo-spellcheck: 0.12.2 -> 0.12.3

AndersonTorres committed:
Merge pull request #211791 from athre0z/zydis-split-zycore

zydis: split zycore into separate derivation

AndersonTorres committed:
Merge pull request #211656 from r-ryantm/auto-update/fvwm3

fvwm3: 1.0.5 -> 1.0.6a

alyssais committed:
fossil: fix cross

Co-authored-by: Artturi <Artturin@artturin.com>

Atemu commented:
Squashed; ready for merge now.
